dask/bag/core.py

```diff
@@ -1014,7 +1014,12 @@ def to_dataframe(self, columns=None):
         columns : list or pandas.DataFrame, optional
             If a list, provides the desired column names. If a
             ``pandas.DataFrame``, it should mirror the column names and dtypes
-            of the output dataframe. If not provided, these will be computed.
+            of the output dataframe. If not provided or provided as list,
+            these will be computed by taking a single element from the first
+            partition, triggering a potentially expensive call to ``compute``.
+            Providing a list is only useful for selecting subset of columns,
+            to avoid an internal compute call you must provide a
+            ``pandas.DataFrame``.
```

Comment:
Yes, i'll change the order and add some more explanations tomorrow.

1422 doesn't change dask/bag/core.py currently, and I'm not sure it'll require a change because `bag.to_dataframe` always creates a full `pd.DataFrame` metadata.